### PR TITLE
Sam Berk Octos 

### PR DESCRIPTION
**Pull request by samanthaberk**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
How did you go about exploring the Edamam API, how did you try querying the API? | I read through the docs and pasted their examples into Postman
Describe your API Wrapper.  How did you decide on the methods you created? | My wrapper has a method to list all recipes matching a search query and a method to display the details for a single recipe. Based on the project requirements I felt this was all that was needed.
Describe an edge case or failure case test you wrote for your API Wrapper. | If the id is not valid (i.e. an empty string), it should throw an error
Explain how VCR aids in testing an API. | It saves a previous call so that I don't need to keep calling the API. This prevents the app from using up calls, makes sure the test doesn't fail erroneously if the API site is down, and ensures I don't need to be connected to the internet to run my tests.
What is the Heroku URL of your deployed application? | https://muncher-api-app.herokuapp.com/
Provide a link to the Trello board you used | https://trello.com/b/Ndbn05au/api-muncher

### CODE REVIEW

File: test/lib/edemam_wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require 'test_helper'
```
Comment (CheezItMan) [inline test/lib/edemam_wrapper.rb:1]: This filename should be `edamam_wrapper_test.rb`, it won't run the tests automatically.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    <%= stylesheet_link_tag    "application" %>
+    <script src="https://developer.edamam.com/attribution/badge.js"></script>
+    <%= javascript_include_tag "application", 'data-turbolinks-track' => true %>
+    <%= csrf_meta_tags %>
+
+  </head>
+
+  <body>
+    <body>
```
Comment (CheezItMan) [inline app/views/layouts/application.html.erb:17]: `body` inside `body`?

File: app/views/recipes/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<section class="recipes-list">
+  <h3>Recipes </h3>
+
+  <% if @recipes %>
+  <ul class="row">
+    <% @recipes.each do |recipe| %>
+    <li class="small-1 medium-2 thumbnail">
```
Comment (CheezItMan) [inline app/views/recipes/index.html.erb:7]: You're using Foundation Grid, neat!

File: app/views/recipes/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      <div class="image">
+        <%= image_url(recipe) %>
```
Comment (CheezItMan) [inline app/views/recipes/index.html.erb:9]: Making the recipe images links would improve the user experience.

File: lib/edemam_wrapper.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        )
+        recipes_list << recipe
+      end
+    end
+    return recipes_list
+  end
+
+  def self.show_recipe(id)
+    encode = URI.encode("#{BASE_URL}?r=#{SINGLE_RECIPE_URL}#{id}&app_id=#{APP_ID}&app_key=#{APP_KEY}")
+    response = HTTParty.get(encode)
```
Comment (CheezItMan) [inline lib/edemam_wrapper.rb:34]: You need to do some testing on the response to ensure that the URI worked, if statements would be appreciated.

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Core Requirements** | 
Git hygiene | Good number of commits and good commit messages
Comprehension questions | Check
**General** | 
Rails fundamentals (RESTful routing, use of named paths) | Your route are **not** RESTful.  You should have the recipe list page on the `/recipes` path and an individual recipe page on `/recipes/:id`, or something similar.  
Semantic HTML | You have a `body` element nested inside a `body` element.  
Errors are reported to the user | No errors are reported for empty search results, and it crashes on a show action with an incorrect URI.
API Wrapper to handle the API requests | Check
Controller testing | **MISSING**
Lib testing | Check
Search Functionality | Check
List Functionality | Check
Show individual item functionality (link to original recipe opens in new tab) |  Check
**Styling** | 
Responsive layout | The search results are responsive, the show page is not.
List View shows 10 items at a time/pagination | Check
The app is styled to create an attractive user interface | The styling looks good.  
**API Features** | 
The App attributes Edaman | Check
The VCR casettes do not contain the API key | Check
**External Resources** | 
Link to Trello Board | The link's there, but I don't have access.
Link to deployed app on Heroku | Check
**Overall** | As I'm sure you know you didn't get the controller testing.  I'm happy to go over it with you this week or next if you like.  You did however get the API functionality working and did well with the styling.  Check out my comments and let me know if you have any questions.
